shell/testing/tester_main.cc

```diff
@@ -513,6 +524,84 @@ int RunTester(const flutter::Settings& settings,
   return completion_observer.GetExitCodeForLastError();
 }
 
+#ifdef _WIN32
+#define EXPORTED __declspec(dllexport)
+#else
+#define EXPORTED __attribute__((visibility("default")))
+#endif
+
+extern "C" {
+EXPORTED Dart_Handle LoadLibraryFromKernel(const char* path) {
+  std::shared_ptr<fml::FileMapping> mapping =
+      fml::FileMapping::CreateReadOnly(path);
+  if (!mapping) {
+    FML_LOG(ERROR) << "Failed to load file at \"" << path << "\".";
+    return Dart_Null();
+  }
+  return DartIsolate::LoadLibraryFromKernel(mapping);
+}
+
+EXPORTED Dart_Handle LookupEntryPoint(const char* uri, const char* name) {
+  if (!uri || !name) {
+    return Dart_Null();
+  }
+  return Dart_GetField(Dart_LookupLibrary(Dart_NewStringFromCString(uri)),
+                       Dart_NewStringFromCString(name));
+}
+
+EXPORTED void Spawn(const char* entrypoint, const char* route) {
+  auto shell = g_shell->get();
+  auto spawn_task = [shell, entrypoint, route]() {
+    auto configuration = RunConfiguration::InferFromSettings(
+        shell->GetSettings(), /*io_worker=*/nullptr,
+        /*launch_type=*/IsolateLaunchType::kExistingGroup);
+    configuration.SetEntrypoint(entrypoint);
```

Comment:
It seems `entrypoint` is the name of the dart function that was marked as `@pragma('vm:entry-point')`. How does this spawn function which library to find that `entrypoint` in?

Should this not use `configuration.SetEntrypointAndLibrary()`?

(The isolate group will have one root library, which is the initial program is was launched from)

Also: If we already have `Spawn` taking the entrypoint by name, why do we need a separate LookupEntryPoint?

Reply:
Spawn is only good for the current library.

Slava's refactoring of the test running harness required being able to lookup other arbitrary entrypoints in other kernels.

Reply:
So @mraleph gave me more context into his code: The `route` is basically a proxy of the actual library's entrypoint to run. So the `Spawn` will spawn an isolate running the same `main` as the initial isolate and distinguishes by the arguments to main what to do.

I'll let @mraleph have review it, as he's more context (that's not included in this CL)